mlflow/models/dependencies_schema.py

```diff
@@ -0,0 +1,162 @@
+        text_column: The name of the text column to use for the embeddings.
+        doc_uri: The name of the column that contains the document URI.
+        other_columns: A list of other columns that are part of the vector index
+                          that need to be retrieved during trace logging.
+        Note: Make sure the text column specified is in the index.
+
+        Example:
+
+        .. code-block:: python
+
+            from mlflow.models import set_vector_search_schema
+
+            set_vector_search_schema(
+                primary_key="chunk_id",
+                text_column="chunk_text",
+                doc_uri="doc_uri",
+                other_columns=["title"],
+            )
+    """
+    globals()[DATABRICKS_VECTOR_SEARCH_PRIMARY_KEY] = primary_key
+    globals()[DATABRICKS_VECTOR_SEARCH_TEXT_COLUMN] = text_column
+    globals()[DATABRICKS_VECTOR_SEARCH_DOC_URI] = doc_uri
+    globals()[DATABRICKS_VECTOR_SEARCH_OTHER_COLUMNS] = other_columns or []
+
+
+def _get_vector_search_schema():
+    """
+    Get the vector search schema defined by the user.
+
+    Returns:
+        VectorSearchIndex: The vector search index schema.
+    """
+    return VectorSearchIndexSchema(
+        name="vector_search_index",
+        primary_key=globals().get(DATABRICKS_VECTOR_SEARCH_PRIMARY_KEY),
+        text_column=globals().get(DATABRICKS_VECTOR_SEARCH_TEXT_COLUMN),
+        doc_uri=globals().get(DATABRICKS_VECTOR_SEARCH_DOC_URI),
+        other_columns=globals().get(DATABRICKS_VECTOR_SEARCH_OTHER_COLUMNS),
+    )
+
+
+class DependenciesType(Enum):
+    """
+    Enum to define the different types of dependencies of the model.
+    """
+
+    VECTOR_SEARCH_INDEX = ResourceType.VECTOR_SEARCH_INDEX.value
+
+
+@dataclass
+class Schema(ABC):
+    """
+    Base class for defining the resources needed to serve a model.
+
+    Args:
+        type (DependenciesType): The type of the schema.
+    """
+
+    type: DependenciesType
+
+    @abstractmethod
+    def to_dict(self):
+        """
+        Convert the resource to a dictionary.
+        Subclasses must implement this method.
+        """
+
+    @classmethod
+    @abstractmethod
+    def from_dict(cls, data: Dict[str, str]):
+        """
+        Convert the dictionary to a Resource.
+        Subclasses must implement this method.
+        """
+
+
+@dataclass
+class VectorSearchIndexSchema(Schema):
+    """
+    Define vector search index resource to serve a model.
+
+    Args:
+        name (str): The name of the vector search index schema.
+        primary_key (str): The primary key for the index.
+        text_column (str): The main text column for the index.
+        doc_uri (Optional[str]): The document URI for the index.
+        other_columns (Optional[List[str]]): Additional columns in the index.
+    """
+
+    type: DependenciesType = DependenciesType.VECTOR_SEARCH_INDEX
+    name: str = None
+    primary_key: str = None
+    text_column: str = None
```

Comment:
These fields are required

```suggestion
    name: str
    primary_key: str
    text_column: str
```